### PR TITLE
new crash function analyse_by_crash

### PR DESCRIPTION
**Pull request by linuxqiao**

### CODE REVIEW

File: lib/crash.sh [hunk 1]
```diff
@@ -15,7 +15,114 @@ check_vmcore_file()
 # To Do
 analyse_by_crash()
```
Comment (sztsian) [inline lib/crash.sh]: If you've finished this function, please move it out of To Do section?

**Comment by sztsian** [comment]
And please squash the two commit into one

File: lib/crash.sh (hunk 1 above)
Comment (linuxqiao) [inline lib/crash.sh]: Thanks, i will update.

**Comment by linuxqiao** [comment]
@sztsian  i asked linqing about this "two commits", just because i use 'git merge new' on my local master branch, and this don't affect what.

About # ToDo session, i will fix this on next commit.

File: lib/crash.sh [hunk 2]
```diff
@@ -15,7 +15,114 @@ check_vmcore_file()
...
+}
+
+get_vmcore_path()
+{
+    local path=""
+
+    [ -f "${K_PATH}" ] && path=$(cat "${K_PATH}") || path="${K_DEFAULT_PATH}"
+
+    if [ -f "${K_NFS}"]; then
+        vmcorepath="$(cat "${K_NFS}")${path}" # need update
```
Comment (emmawu) [inline lib/crash.sh]: you need a space before ]

**Comment by emmawu** [comment]
Hi Qiao, before you commit you fix. There is a dummy way to merge your commits into one:
1. git reset HEAD~2   # unstage the changes in your last 2 commits (2ac16fe and 00a9550)
2. git add lib/crash.sh   # stage file crash.sh you've updated.
3. git commit -sm "Add new function: analyse_by_crash"  # commit the change in crash.sh
4. git checkout -- * # discard the changes I made in kdump.sh and clean-up tests. 
5. git push -f # force push your changes to your own kdump-test-1 repo. 

if you'd like to make any change to crash.sh again to fix some errors:
1. #update crash.sh
2. git add lib/crash.sh  # stage the change
3. git commit --amend -C HEAD  # add the change to your last commit
4. git push -f # force push your last commit again to your own repo

Then we'll see this PR right (always one commit).  This is not the typical procedures..This is just a dummy way to get things right if you like to create PR from your own master branch.

**Pull request merged**